gradio/blocks.py

```diff
@@ -456,6 +458,23 @@ def convert_component_dict_to_list(outputs_ids: List[int], predictions: Dict) ->
     return predictions
 
 
+def add_request_to_inputs(
+    fn: Callable, inputs: List[Any], request: routes.Request | List[routes.Request]
+):
+    """
+    Adds the FastAPI Request object to the inputs of a function if the type of the parameter is FastAPI.Request.
+    """
+    param_names = inspect.getfullargspec(fn)[0]
+    try:
+        parameter_types = typing.get_type_hints(fn)
+        for idx, param_name in enumerate(param_names):
+            if parameter_types.get(param_name, "") == routes.Request:
+                inputs.insert(idx, request)
+    except TypeError:  # A TypeError is raised if the function is a partial or other rare cases.
```

Comment:
Should we just raise an error here? If we can't add a request, user will get a "TypeError: fn missing 1 required positional argument: 'request'" or something like that down the line. I wonder if that error would be more actionable if we controlled the exception message, e.g. `Cannot add request to function signature. This typically happens if the function is a partial. Please refactor your function` ?

Reply:
If we were to raise an error here, then it would break support for partial functions altogether since `add_request_to_inputs()` is called in all cases, regardless of whether the user actually supplies a parameter with type hint `gr.Request`. The reason I put the `TypeError` was because `typing.get_type_hints()` raises a `TypeError` when it inspects a partial function, so this way, we simply don't add the request object when working with partials. 

I think that partials lose all typing hint information so it might not be possible to use this approach with partials. I don't think it's that big of a deal, but open to suggestions.